Document/0x05i-Testing-Code-Quality-and-Build-Settings.md

```diff
@@ -484,88 +484,64 @@ There are various steps to take:
 
 ### Overview
 
-Because decompiling Java classes is trivial, applying some basic obfuscation to the release byte-code is recommended. [ProGuard](0x08-Testing-Tools.md#proguard) offers an easy way to shrink and obfuscate code and to strip unneeded debugging information from the byte-code of Android Java apps. It replaces identifiers, such as class names, method names, and variable names, with meaningless character strings. This is a type of layout obfuscation, which is "free" in that it doesn't impact the program's performance.
-
-Since most Android applications are Java-based, they are [immune to buffer overflow vulnerabilities](https://owasp.org/www-community/vulnerabilities/Buffer_Overflow "Java Buffer Overflows"). Nevertheless, a buffer overflow vulnerability may still be applicable when you're using the Android NDK; therefore, consider secure compiler settings.
+The tests used to detect the presence of binary protection mechanisms heavily depend on the language used for developing the application. These are some basic guidelines:
+
+- In general all binaries must be tested, this includes both the app main executable as well as all libraries/dependencies.
+- In some cases the nature of the language they were written will offer sufficient protection (e.g. Java has Garbage Collection similar to ARC on iOS), so even if a feature is not enabled the test may pass.
```

Comment:
```suggestion
- Some languages are inherently protected against certain vulnerability classes. For example Java has Garbage Collection similar to ARC on iOS), so even if a feature is not enabled the test may pass.
```
I rewrote this a bit, but would actually like to remove it. "if a feature is not enabled the test may pass" is weird. The features are just different for those languages. If there is no risk of a memory leak, there are no 'free security features related to memory leaks' for that language so it just doesn't apply.

I think it makes sense to just focus on native code and skip the java stuff.